lnwallet/commitment.go

```diff
@@ -151,9 +154,31 @@ type ScriptInfo struct {
 // CommitScriptToRemote creates the script that will pay to the non-owner of
 // the commitment transaction, adding a delay to the script based on the
 // channel type.
-func CommitScriptToRemote(_ channeldb.ChannelType, csvTimeout uint32,
+func CommitScriptToRemote(chanType channeldb.ChannelType, csvTimeout uint32,
 	key *btcec.PublicKey) (*ScriptInfo, error) {
 
+	// If this channel type has anchors, we derive the delayed to_remote
+	// script.
+	if chanType.HasAnchors() {
```

Comment:
Is this then the thing that prompts the creation of a `CommitmentBuilder` interface?